libs/core/app/src/search/learning-objective-search.ts

```diff
@@ -125,6 +125,11 @@ export const searchLearningObjectives = async ({
 }) => {
   const opts: SearchOptions = {
     fuzzy: 0.2,
+    fields: ps.searchField
+      ? ps.searchField === "all"
+        ? undefined
+        : [ps.searchField]
+      : undefined,
```

Comment:
`ps.searchField` can never be undefined; Zod makes sure of that. So this can simply be: 

```ts
    fields: ps.searchField === "all" ? undefined : [ps.searchField],
```

The reason why the field search here was a placebo is because the tokenizer breaks apart the learning objectives on "dots" making the search via ID completely useless (010.10, will return any LO that contains `010` and `01`.

On my WIP I was able to improve this by doing:

```ts
const SEARCH_INDEX = new MiniSearch<SearchDocument>({
  fields: [/** **/],
  storeFields: ["id"] satisfies SearchField[],
  tokenize: (s, fieldName) => {
    if (fieldName === "learningObjectives") return s.split(", ");
    return MiniSearch.getDefault("tokenize")(s, fieldName);
  },
});
``` 

and then for the `opts`

```ts
  const opts: SearchOptions = {
    fuzzy: idSearchFields.includes(ps.searchField) ? false : 0.2,
    prefix: idSearchFields.includes(ps.searchField),
    fields: ps.searchField === "all" ? undefined : [ps.searchField],
    tokenize: (s) => {
      if (ps.searchField === "learningObjectives") return s.split(", ");
      return MiniSearch.getDefault("tokenize")(s);
    },
  };
```

Long story short, the goal is to use Fuzzy search for text fields, and prefix  search(ie.: exact matches) for ID fields. it's also vital that for learningObjectives, and IDS the tokenizer does not split on "."

This is mostly non actionable right now, just sharing in case you are interested in improving the search a bit more in the future :D

Reply:
Oh alright, I was wondering why for some terms the LO search gives good results and sometimes they're barely related. Makes sense